PitchShiftEffect: add description comments

davidchocholaty opened the pull request:
This PR adds the comments, primarily for the usage
of the RubberBand API, for the Pitch shift effect processing.

davidchocholaty committed:
PitchShiftEffect: add description comments

This commit adds the comments, primarily for the usage
of the RubberBand API, for the Pitch shift effect processing.

daschuer commented on src/effects/backends/builtin/pitchshifteffect.cpp:
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
             m_currentFormant != formantPreserving) {
         m_currentFormant = formantPreserving;
 
+        // Set the RubberBand option for the formant preserving.
```
This comment just repeats the code.

daschuer commented on src/effects/backends/builtin/pitchshifteffect.cpp:
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
     }
 
+    // Get the parameter value of the Pitch knob
+    // based on the current setting of the Range knob.
```
This comment does also not add any value. 
You may explain that one of from 0...X and the other from -x to +y ... and describe the implications.

daschuer commented on src/effects/backends/builtin/pitchshifteffect.cpp:
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
         pitchParameter = roundToFraction(pitchParameter, kSemitonesPerOctave);
     }
 
+    // Calculate the result pitch setting for the RubberBand.
```
Here also no value. You may tell us why std::pow(2.0 is required.

daschuer reviewed (changes requested):
I have left some comments.  Sorry if they look rude, that was not meant like that. 
I am happy about such Document PRs, Thank you.  

It is just, that I share the opinion that comments are only a last resort to explain difficult code parts in cases it was not possible to put the information into the code itself.

That helps that the reader does not waste time reading it and limit the risk that the code comment is rotting after refactoring.  

In this case, think about the pitfalls you have managed and write about them.

davidchocholaty commented:
Absolutely no problem, thank you very much for the advice and proposals. I will do it this way.

davidchocholaty committed:
PitchShiftEffect: remove unnecessary comment

This commit removes the unnecessary comment of the setting the formant
preserving for the RubberBand processing based on that it just repeats
the code.

davidchocholaty committed:
PitchShiftEffect: explain why the pow is required

The commit changes the comment of the pitch value setting
with a more detailed explanation of why the pow is needed,
and the formula of the calculation is added.

davidchocholaty committed:
PitchShiftEffect: add range parameter explanation

The commit adds a closer explanation of how the resulting pitch value
is changing based on the range value with a couple of real examples.

davidchocholaty committed:
PitchShiftEffect: add semitones mode description

The commit adds the description of rounding to fraction to keep
the resulting pitch value in the semitone chromatic scale.

davidchocholaty committed:
PitchShiftEffect: remove unuseful comments

The commit removes unuseful comments about retrieving data
from RubberBand.

daschuer reviewed (approved):
Thank you for all the additional details :-)

davidchocholaty commented:
Thank you for merging. :+1:
